src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -280,9 +281,11 @@ public Document GetRequiredDocument()
                 (workspace, solution) = await lspWorkspaceManager.GetLspSolutionInfoAsync(cancellationToken).ConfigureAwait(false);
             }
 
-            if (workspace is null)
+            if (workspace is null || solution is null)
             {
                 logger.LogError($"Could not find appropriate workspace for operation {workspace} on {method}");
+                FatalError.ReportWithDumpAndCatch(new Exception(
+                    $"Could not find appropriate workspace or solution for operation {workspace}, {solution} on {method}"), ErrorSeverity.General);
```

Comment:
These strings seem funky since if this is null, those won't really contain anything useful.